core/slice.go

```diff
@@ -467,13 +467,21 @@ func (sl *Slice) UpdateDom(oldTerminus common.Hash, pendingHeader types.PendingH
 	}
 }
 
-func (sl *Slice) randomRelayArray() [3]int {
+func (sl *Slice) randomRelayArray() []int {
 	rand.Seed(time.Now().UnixNano())
-	nums := [3]int{0, 1, 2}
+
+	// Initialize a slice of size common.Width with integers from 0 to common.Width-1
+	nums := make([]int, common.Width)
+	for i := 0; i < common.Width; i++ {
+		nums[i] = i
+	}
+
+	// Shuffle the nums slice using the Fisher-Yates shuffle algorithm
 	for i := len(nums) - 1; i > 0; i-- {
```

Comment:
we can probably use the rand.Shuffle function here, probably cleaner

Reply:
done